src/share/poudriere/image.sh

```diff
@@ -130,10 +135,83 @@ mkminiroot() {
 	gzip -9 ${OUTPUTDIR}/${IMAGENAME}-miniroot
 }
 
+create_zfs_be_datasets() {
+	local OPT OPTSTR
+
+	if [ -n "$1" ]; then
+		. $1
+	fi
+	: ${ZFS_DATASETS:="
+		# DATASET	OPTIONS (space separated)
+
+		# Boot Environment [BE] root and default boot dataset
+		/$ZFS_BEROOT_NAME			mountpoint=none
+		/$ZFS_BEROOT_NAME/$ZFS_BOOTFS_NAME	mountpoint=/
+
+		# Compress /tmp, allow exec but not setuid
+		/tmp		mountpoint=/tmp exec=on setuid=off
+
+		# Don't mount /usr so that 'base' files go to the BEROOT
+		/usr		mountpoint=/usr canmount=off
+
+		# Home directories separated so they are common to all BEs
+		/usr/home	# NB: /home is a symlink to /usr/home
+
+		# Ports tree
+		/usr/ports	setuid=off
+
+		# Source tree (compressed)
+		/usr/src
+
+		# Create /var and friends
+		/var		mountpoint=/var canmount=off
+		/var/audit	exec=off setuid=off
+		/var/crash	exec=off setuid=off
+		/var/log	exec=off setuid=off
+		/var/mail	atime=on
+		/var/tmp	setuid=off
+	"}
+
+	msg "Creating ZFS Datasets"
+	echo "$ZFS_DATASETS" | while read dataset options; do
```

Comment:
I suspect there's an elegant `awk` or `sed` version of this. It's fine as is though.